PWGCF/GenericFramework/Tasks/flowGenericFramework.cxx

```diff
@@ -278,163 +320,183 @@ struct GenericFramework {
+        QAregistry.fill(HIST("phi_gen"), phi);
+        QAregistry.fill(HIST("eta_gen"), eta);
+        QAregistry.fill(HIST("pt_gen"), pt);
       }
+    } else {
+      pt = track.pt();
+      eta = track.eta();
+      phi = track.phi();
+      if (cfgFillQA) {
+        QAregistry.fill(HIST("phi"), phi);
+        QAregistry.fill(HIST("eta"), eta);
+        QAregistry.fill(HIST("pt_dcaXY_dcaZ"), pt, track.dcaXY(), track.dcaZ());
+      }
+      if (cfgFillWeights)
+        fWeights->Fill(phi, eta, vtxz, pt, centrality, 0);
+      if (!setCurrentParticleWeights(weff, wacc, phi, eta, pt, vtxz))
+        return;
     }
-    for (uint l_ind = 0; l_ind < corrconfigs.size(); l_ind++) {
-      FillFC(fFC, corrconfigs.at(l_ind), centrality, l_Random);
-    }
+    bool WithinPtPOI = (ptpoilow < pt) && (pt < ptpoiup); // within POI pT range
+    bool WithinPtRef = (ptreflow < pt) && (pt < ptrefup); // within RF pT range
+    if (WithinPtRef)
+      fGFW->Fill(track.eta(), fPtAxis->FindBin(pt) - 1, phi, wacc * weff, 1);
+    if (WithinPtPOI)
+      fGFW->Fill(track.eta(), fPtAxis->FindBin(pt) - 1, phi, wacc * weff, 2);
+    if (WithinPtPOI && WithinPtRef)
+      fGFW->Fill(track.eta(), fPtAxis->FindBin(pt) - 1, phi, wacc * weff, 4);
+    registry.fill(HIST("phi_eta_vtxZ_corrected"), phi, eta, vtxz, wacc);
   }
 
   void processRawData(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::Mults, aod::CentFT0Cs>>::iterator const& collision, aod::BCsWithTimestamps const&, myTracks const& tracks)
   {
+    if (!collision.sel8())
+      return;
     const auto centrality = collision.centFT0C();
     auto bc = collision.bc_as<aod::BCsWithTimestamps>();
-    if (!collision.sel8())
+    if (cfgUse22sEventCut && !eventSelected(collision, tracks.size(), centrality))
       return;
     QAregistry.fill(HIST("globalTracks_PVTracks"), collision.multNTracksPV(), tracks.size());
+    QAregistry.fill(HIST("cent_nch"), tracks.size(), centrality);
     loadCorrections(bc.timestamp());
-    processData<kRaw>(collision, tracks, centrality);
+    processCollision<kRaw>(collision, tracks, centrality);
   }
   PROCESS_SWITCH(GenericFramework, processRawData, "Process analysis for derived data", true);
 
+  Filter dcaFilter = nabs(aod::track::dcaXY) < cfgDCAxy;
+  void processReco(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::CentFT0Cs>>::iterator const& collision, aod::BCsWithTimestamps const&, soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksDCA, aod::McTrackLabels>> const& tracks, aod::McParticles const&)
+  {
+    const auto centrality = collision.centFT0C();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    if (!collision.sel8())
+      return;
+    loadCorrections(bc.timestamp());
+    processCollision<kRaw>(collision, tracks, centrality);
+  }
+  PROCESS_SWITCH(GenericFramework, processReco, "Process analysis for MC reconstructed events", false);
```

Comment:
Have in mind that with the signature of this `processReco` function it will not be able to process reconstructed data. It is being required that `tracks` had joined the `McTrackLabels` column which will be not available for reconstructed data. Or was the reconstructed data intended for the `processRawData` function? If so I was confused by the `SWITCH` comment